src/plugins/crypto/README.md

```diff
@@ -18,9 +18,33 @@ It also needs to be decrypted whenever an admissible access (read) is being perf
 The users of Elektra should not be bothered too much with the internals of the cryptographic operations.
 Also the cryptographic keys must never be exposed to the outside of the crypto module.
 
+The crypto plugin supports different libraries as provider for the cryptographic operations.
+At the moment the following crypto APIs are supported:
+
+- OpenSSL (libcrypto)
+- libgcrypt
+
 ## Dependencies ##
 
+#ifdef ELEKTRA_CRYPTO_API_GCRYPT
 - `libgcrypt20-dev` or `libgcrypt-devel`
+#endif
+#ifdef ELEKTRA_CRYPTO_API_OPENSSL
+- `libssl-dev` or `openssl-devel`
+#endif
+
+## How to compile ##
+
+The following compile variants are available:
+
+1. crypto_gcrypt
+2. crypto_openssl
+
+Add the variants, that you want (you can add one of them or all), to the `PLUGINS` variable in `CMakeCache.txt` and re-run `cmake`.
+
+To add all variants the variable contains at least:
+
+	PLUGINS=crypto;crypto_gcrypt;crypto_openssl
```

Comment:
We could also have an CRYPTO variable that gets replaced to `crypto;crypto_gcrypt;crypto_openssl` (similar to SWIG)